core/src/repair/quic_endpoint.rs

```diff
@@ -249,44 +371,46 @@ async fn run_server(
                 ));
             }
             Err(error) => {
-                debug!("Error while accepting incoming connection: {error:?} from {remote_addr}");
+                error!("Error while accepting incoming connection: {error:?} from {remote_addr}");
             }
         }
     }
     report_metrics_task.abort();
 }
 
-async fn run_client(
+async fn run_client<T>(
     endpoint: Endpoint,
-    mut receiver: AsyncReceiver<LocalRequest>,
-    remote_request_sender: Sender<RemoteRequest>,
+    client_name: &'static str,
+    mut receiver: AsyncReceiver<(SocketAddr, Bytes)>,
+    sender: Sender<T>,
     bank_forks: Arc<RwLock<BankForks>>,
     prune_cache_pending: Arc<AtomicBool>,
-    router: Arc<AsyncRwLock<HashMap<SocketAddr, AsyncSender<LocalRequest>>>>,
+    router: Arc<AsyncRwLock<HashMap<SocketAddr, AsyncSender<Bytes>>>>,
     cache: Arc<Mutex<HashMap<Pubkey, Connection>>>,
-) {
+) where
+    T: 'static + From<(Pubkey, SocketAddr, Bytes)> + Send,
+{
     let stats = Arc::<RepairQuicStats>::default();
-    let report_metrics_task =
-        tokio::task::spawn(report_metrics_task("repair_quic_client", stats.clone()));
-    while let Some(request) = receiver.recv().await {
-        let Some(request) = try_route_request(request, &*router.read().await, &stats) else {
+    let report_metrics_task = tokio::task::spawn(report_metrics_task(client_name, stats.clone()));
+    while let Some((remote_address, bytes)) = receiver.recv().await {
+        let Some(bytes) = try_route_bytes(&remote_address, bytes, &*router.read().await, &stats)
+        else {
             continue;
         };
-        let remote_address = request.remote_address;
         let receiver = {
             let mut router = router.write().await;
-            let Some(request) = try_route_request(request, &router, &stats) else {
+            let Some(bytes) = try_route_bytes(&remote_address, bytes, &router, &stats) else {
                 continue;
             };
             let (sender, receiver) = tokio::sync::mpsc::channel(ROUTER_CHANNEL_BUFFER);
-            sender.try_send(request).unwrap();
+            sender.try_send(bytes).unwrap();
             router.insert(remote_address, sender);
```

Comment:
When is it removed from the router?

Reply:
In `fn handle_connection` right after dropping the connection.
Also in `fn prune_connection_cache` we drop entries where the sender is already closed.